rust/core-lib/src/tabs.rs

```diff
@@ -133,6 +135,14 @@ pub struct DocumentCtx {
     update_channel: mpsc::Sender<(ViewIdentifier, PluginUpdate, usize)>
 }
 
+#[derive(Serialize, Deserialize)]
+/// A request for measuring the widths of strings all of the same style (a request
+/// from core to front-end).
+pub struct WidthReq {
```

Comment:
Nope that sounds reasonable. The other place it could live is in `rpc.rs`, which contains all the types for the client->core protocol, but doesn't currently have anything to do with the core->client side. Post refactor it could also be in `client.rs`.